MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/turf/WithinExpressionActivity.kt

```diff
@@ -0,0 +1,266 @@
+import android.os.PersistableBundle
+import androidx.appcompat.app.AppCompatActivity
+import com.mapbox.geojson.*
+import com.mapbox.mapboxsdk.camera.CameraPosition
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.maps.MapView
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.mapboxsdk.style.expressions.Expression.within
+import com.mapbox.mapboxsdk.style.layers.CircleLayer
+import com.mapbox.mapboxsdk.style.layers.FillLayer
+import com.mapbox.mapboxsdk.style.layers.LineLayer
+import com.mapbox.mapboxsdk.style.layers.Property.NONE
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory.*
+import com.mapbox.mapboxsdk.style.layers.SymbolLayer
+import com.mapbox.mapboxsdk.style.sources.GeoJsonOptions
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource
+import com.mapbox.mapboxsdk.testapp.R
+import kotlinx.android.synthetic.main.activity_physical_circle.*
+
+/**
+ * An Activity that showcases the within expression to filter features outside a geometry
+ */
+class WithinExpressionActivity : AppCompatActivity() {
+
+  private lateinit var mapboxMap: MapboxMap
+  private val handler: Handler = Handler()
+  private val runnable: Runnable = Runnable {
+    optimizeStyle()
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_within_expression)
+
+    mapView.onCreate(savedInstanceState)
+    mapView.getMapAsync { map ->
+      mapboxMap = map
+
+      // Setup camera position above Georgetown
+      mapboxMap.cameraPosition = CameraPosition.Builder()
+        .target(LatLng(38.90628988399711, -77.06574689337494))
+        .zoom(14.5)
+        .build()
+
+      // Wait for the map to become idle before manipulating the style and camera of the map
+      mapView.addOnDidBecomeIdleListener(object : MapView.OnDidBecomeIdleListener {
+        override fun onDidBecomeIdle() {
+          handler.postDelayed(runnable, 2500)
+          mapView.removeOnDidBecomeIdleListener(this)
+        }
+      })
+
+      // Load mapbox streets and add lines and circles
+      setupStyle()
+    }
+  }
+
+  private fun setupStyle() {
+    // Assume the route is represented by an array of coordinates.
+    val coordinates = listOf<Point>(
+      Point.fromLngLat(
+        -77.06866264343262,
+        38.90506061276737
+      ),
+      Point.fromLngLat(
+        -77.06283688545227,
+        38.905194197410545
+      ),
+      Point.fromLngLat(
+        -77.06285834312439,
+        38.906429843444094
+      ),
+      Point.fromLngLat(
+        -77.0630407333374,
+        38.90680554236621
+      )
+    )
+
+    // Convert to a LineString
+    val routeLineString = LineString.fromLngLats(coordinates)
+
+    // Create buffer around linestring
+    val bufferedRouteGeometry = bufferLineStringGeometry(routeLineString)
+
+    // Setup style with custom layers,
```

Comment:
Why does this say "custom layers" here? Seems a bit confusing with the `MGLOpenGLStyleLayer` custom layers

Reply:
custom layers on top mapbox streets, I will change the naming ;)